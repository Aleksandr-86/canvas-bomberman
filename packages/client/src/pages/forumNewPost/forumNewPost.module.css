.forum {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.forumWrapper {
  display: flex;
  flex-direction: column;
}

.forumLink {
  max-width: 250px;
  margin-bottom: 30px;
  padding: 10px 50px;
  border: 5px solid black;
  border-radius: 15px;
  background: hsl(29deg 6% 89%);
  color: black;
  font-family: 'Rubik Bubbles', fantasy;
  text-transform: uppercase;
  cursor: pointer;
  transition: all 0.2s linear;
}

.forumTitle {
  font-weight: 400;
  font-size: 44px;
  font-family: 'Rubik Bubbles', fantasy;
  line-height: 49px;
  text-align: center;
}
