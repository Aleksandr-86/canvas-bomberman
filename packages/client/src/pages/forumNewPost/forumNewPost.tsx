import { Button } from '../../components/button/button'
import { Link } from 'react-router-dom'
import classes from './forumNewPost.module.css'
import { Input } from '../../components/input/input'

export const ForumNewPost = () => {
  return (
    <div className={classes.forum}>
      <h2 className={classes.forum__title}>Новая тема</h2>

      <div className={classes.forum__wrapper}>
        <form>
          <Input type="text" placeholder="Название темы" />
          <Button
            type="submit"
            className="forum__btn--create">создать тему</Button>
        </form>
        <Link to="/forum" className={classes.forum__link}>
          назад
        </Link>
      </div>
    </div >
  )
}
