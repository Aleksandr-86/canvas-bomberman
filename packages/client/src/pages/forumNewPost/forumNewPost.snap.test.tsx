import * as renderer from 'react-test-renderer'
import { ForumNewPost } from './forumNewPost'
import { MemoryRouter } from 'react-router-dom'

describe(`Snapshot test page ForumNewPost`, () => {
  test(`Should correct render (without props)`, () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={[`/forum-new-post`]}>
          <ForumNewPost />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
