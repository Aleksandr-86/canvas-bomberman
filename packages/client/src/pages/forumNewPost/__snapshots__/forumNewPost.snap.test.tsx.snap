// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForumNewPost/> snapshot test cases <ForumNewPost/> should render <div className=forum> with child elements 1`] = `
<div
  className="forum"
>
  <h2
    className="forumTitle"
  >
    Новая тема
  </h2>
  <div
    className="forumWrapper"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        <input
          className="form__input"
          name="body"
          placeholder="Название темы"
          type="text"
        />
      </label>
      <button
        className="button forumBtnCreate"
        type="submit"
      >
        создать тему
      </button>
    </form>
    <a
      className="forumLink"
      href="/forum"
      onClick={[Function]}
    >
      назад
    </a>
  </div>
</div>
`;
